### PR TITLE
Dark Days of the Dead: Headshots, deadly bites, and a cleanup of out-of-setting stuff.

### PR DESCRIPTION
**Pull request by I-am-Erk**
#### Summary
Mods "Dark Days of the Dead: Headshot requirements and deadly zombie bites are in!"

#### Purpose of change
Dark Days of the Dead is meant to turn CDDA into a Romero zombie simulation. I have been behind on keeping this up to date.

#### Describe the solution
Uses the Weakpoints system to make headshot attacks far more powerful against zombies, and non-headshot attacks much weaker.  Sufficient severe damage can destroy the zombie, currently set as 7x their default hitpoints. You can still whack 'em in the arms and legs to knock them down or stagger them to slow them down (although this does even less damage, because it's hard to mangle a zombie like that). If we ever work out how to use weakpoints to do limb removal I may see about adding that somehow. You may have a chance with a swinging melee weapon of permanently knocking them down, which is the best I can do to represent destroying the leg. You can also stagger or knockdown a zombie even if you don't damage them. I may need to review the knockdown effect and make sure it doesn't totally invalidate the zombies.

This does make headshots easier to get, especially in melee, which I think we can assume is because avatars rapidly learn to aim high. Since non-headshots do little, we're already modeling the reduced hit rate. Plus, a knocked down zombie can be coup-de-graced with an even more damaging headshot that is easier to get.

Also adds a deadly zombie bite, adapted from but not identical to the Deadly Bites challenge mod.

Disables portal storms and removes a number of not-appropriate scenarios.

Turns of zombie animals, after some discussion in discord over whether most people consider them "classic" or not.

#### Testing
-Zombies can be killed
-Zombies don't bleed
-Headshots are still tough to get, but even an unskilled character can get them
-The damage from a headshot is often enough to take a zombie down pretty fast... but often not
-Deadly Bites hard to test, will probably need longer term playtesting.
-knockdowns and staggers are more common and you get a lot more headshots on a downed Z
-At high skill, you hit the head a ton. An experienced warrior will still find it fairly doable to fight.
image

#### Additional context

This should fixes a bug where weakpoints were not applying effect messages correctly. That should be another PR but I am not patient enough to split it out when it was necessary for this one.

### CODE REVIEW

File: data/mods/classic_zombies/contagion/effect.json [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+      "You've got a painful bite from a zombie.  You know how this ends.",
+      "The area around your zombie bite is festering and red, and you're feeling hot and cold at once.",
+      "Everything's so dark.  You can feel it, you don't have much time left."
+    ],
+    "apply_message": "You've been bitten.  It's bad.",
+    "death_msg": "You succumb to your zombie bite.",
+    "apply_memorial_log": "Died of the zombie plague.",
+    "chance_kill": [ [ -1, 1 ], [ -1, 1 ], [ 1, 100000 ] ],
+    "death_event": "dies_of_infection",
+    "max_intensity": 5,
```
Comment (anothersimulacrum) [inline data/mods/classic_zombies/contagion/effect.json:16]: Not too familiar with effects, but it looks like this only has 3 stages.

Comment (I-am-Erk) [reply data/mods/classic_zombies/contagion/effect.json:16]: woops, forgot to fix that one, thanks

File: data/mods/classic_zombies/monsters/weakpoints.json [hunk 2]
```diff
@@ -0,0 +1,444 @@
...
+        "effects": [
+          {
+            "effect": "blind",
+            "duration": [ 1, 2 ],
+            "chance": 1,
+            "message": "The %s is blinded!  The goggles, they did nothing!",
+            "damage_required": [ 20, 100 ]
+          }
+        ],
+        "//": "All right, I'll admit it, I only added this effect so I could make the joke.  What you gonna do about it?"
```
Comment (TheShadowFerret) [inline data/mods/classic_zombies/monsters/weakpoints.json:417]: *Deletes weakpoint*
:3

**Review by TheShadowFerret — COMMENTED** [review]
``They can be destroyed with catastrophic damage, by completely destroying their bodies (represented by depleting their very large HP pool) or by a headshot.``

Will the former method be removed by the design doc, or is now meant to be accomplished by  other methods(fire I assume)?

**Comment by I-am-Erk** [comment]
I would like to still have it possible to deal catastrophic damage to them to kill them, when that is possible.

For the moment, this is running into some weird behaviour. I believe the system is working correctly, but we still get a "damage done" message. The zombie HP doesn't go down. Zombies in a fight now seem to bleed to death, which means I will need to disable zombie bleeding for this step as well. However, the "damage done" message needs to be removed.

The problem may be that my test weapon is using sweep attack specials though, I am not sure if that messes with weakpoints, and if it does, I am not sure how to work around that.

**Pull request merged**